### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Skywalker-11**
Applied most of the advisories of Scrutinizer which mainly consists of the following changes:
- refactor methods to reduce duplication
- split html_start function to multiple sub functions for better overview, more modular code
- initialize variables
- remove unused code
- adjust parameter comments to reflect the actually used types
- small fixes

Our code quality gets better and better just compare 
[https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/master](https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/master) (code from 02.01.17)
[https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/develop](https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/develop) (current develop)
[https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/refactoring](https://scrutinizer-ci.com/g/Skywalker-11/1.2.0/issues/refactoring) (with this pr)

### CODE REVIEW

File: mailscanner/functions.php [hunk 1]
```diff
@@ -352,172 +397,173 @@ function html_start($title, $refresh = 0, $cacheable = true, $report = false)
...
-                        $percent = '<span style="color:red">';
-                    }
-                    $percent .= ' [';
-                    $percent .= round($free_space / $total_space, 2) * 100;
-                    $percent .= '%] ';
-                    $percent .= '</span>';
-                    echo '    <tr><td>' . $disk['mountpoint'] . '</td><td colspan="2" align="right">' . formatSize($free_space) . $percent . '</td>' . "\n";
-                }
+            if ($pqerror !== '') {
+                echo '    <tr><td colspan="3">Warning: An error occured:' . $pqerror . '</td>' . "\n";
```
Comment (stefaweb) [inline mailscanner/functions.php]: `Warning: An error occured:`

To translate?

Comment (Skywalker-11) [inline mailscanner/functions.php]: Was there before but I can do it

Comment (stefaweb) [inline mailscanner/functions.php]: We need to add a space after "occurred:".

`echo '    <tr><td colspan="3">Warning: An error occured:' .  ' ' . $pqerror . '</td>' . "\n";`

Comment (Skywalker-11) [inline mailscanner/functions.php]: @stefaweb what is the fr translation?

Comment (stefaweb) [inline mailscanner/functions.php]: `Attention : une erreur s\'est produite :`

File: tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php [hunk 2]
```diff
@@ -79,7 +79,7 @@
                             $output[$msgid]['message'] = 'N/A';
                             while (!@feof($fh)) {
                                 if ($line = @fgets($fh, 1024)) {
-                                    switch (true) {
+                                    switch (1) {
```
Comment (endelwar) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]: hat tip for this!

File: mailscanner/functions.php [hunk 3]
```diff
@@ -1003,7 +1040,7 @@ function __($string)
 /**
  * Returns true if $string is valid UTF-8 and false otherwise.
  *
- * @param $string
+ * @param string
```
Comment (endelwar) [inline mailscanner/functions.php]: `@param string $string`

File: mailscanner/functions.php [hunk 4]
```diff
@@ -508,7 +508,7 @@ function printMTAQueue()
                 }
             }
             if ($pqerror !== '') {
-                echo '    <tr><td colspan="3">Warning: An error occured:' . $pqerror . '</td>' . "\n";
+                echo '    <tr><td colspan="3">' . __('errorWarning03') . $pqerror . '</td>' . "\n";
```
Comment (stefaweb) [inline mailscanner/functions.php]: Add a space between __('errorWarning03') and $pqerror.

**Comment by endelwar** [comment]
That is a big improvement of code base! Great work @Skywalker-11

File: tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php (hunk 2 above)
Comment (Skywalker-11) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]: well its just what scrutinizer suggested

**Pull request merged**